### PR TITLE
Charge in BeamRelevant Redu. Diagnostic

### PR DESCRIPTION
**Pull request by ax3l**
Adds the beam parameter "charge" to our reduced diagnostics, type "BeamRelevant".

Will be useful for libEnsemble studies :)

## To Do

- [x] rebase on top of bug fix in #879

## Follow-Up

- potentially needs multiplication with ionization state `ion_lev` for ion beams?

### CODE REVIEW

File: Examples/Tests/initial_distribution/inputs [hunk 1]
```diff
@@ -176,3 +176,9 @@ h4z.bin_number                           = 50
 h4z.bin_min                              = -1.0
 h4z.bin_max                              = +1.0
 h4z.histogram_function(t,x,y,z,ux,uy,uz) = "z"
+
+# our little beam monitor
+bmmntr.type                              = BeamRelevant
```
Comment (MaxThevenet) [inline Examples/Tests/initial_distribution/inputs:181]: vowels are getting expensive these days :).

Comment (ax3l) [reply Examples/Tests/initial_distribution/inputs:181]: haha, that's why I needed a comment :D

File: Source/Diagnostics/ReducedDiags/BeamRelevant.cpp [hunk 2]
```diff
@@ -101,7 +104,8 @@ BeamRelevant::BeamRelevant (std::string rd_name)
             ofs << "[13]pz_rms(kg*m/s)";  ofs << m_sep;
             ofs << "[14]gamma_rms()";     ofs << m_sep;
             ofs << "[15]emittance_x(m)";  ofs << m_sep;
-            ofs << "[16]emittance_z(m)";  ofs << std::endl;
+            ofs << "[16]emittance_x(m)";  ofs << m_sep;
```
Comment (Yin-YinjianZhao) [inline Source/Diagnostics/ReducedDiags/BeamRelevant.cpp:107]: @ax3l @MaxThevenet Found a typo here, sorry I missed this when reviewing.
`ofs << "[16]emittance_x(m)";  ofs << m_sep;`
should be
`ofs << "[16]emittance_x(z)";  ofs << m_sep;`

Comment (ax3l) [reply Source/Diagnostics/ReducedDiags/BeamRelevant.cpp:107]: oh no! will do a follow-up PR!

Comment (ax3l) [reply Source/Diagnostics/ReducedDiags/BeamRelevant.cpp:107]: fix available in #900 :)

**Pull request merged**